File: SocialPulse/SocialPulse.Core/Entities/Subscription.cs
namespace SocialPulse.Core
{
    public class Subscription : BaseEntity
    {
        public int UserId { get; set; }
        public User User { get; set; } = null!;
        public bool Active { get; set; }
        public DateTime ExpirationDate { get; set; }
    }
}

File: SocialPulse/UI/admin_socialpulse/lib/models/subscription.dart
import 'package:json_annotation/json_annotation.dart';

part 'subscription.g.dart';

@JsonSerializable()
class Subscription {
  int? id;
  bool? active;
  int? userId;
  DateTime? expirationDate;
  DateTime? createdAt;

  Subscription(
      this.id, this.active, this.userId, this.expirationDate, this.createdAt);

  factory Subscription.fromJson(Map<String, dynamic> json) =>
      _$SubscriptionFromJson(json);

  Map<String, dynamic> toJson() => _$SubscriptionToJson(this);
}
